File: projeto 2/academia/aspnet-backend/AcademiaAPI/Models/Matricula.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AcademiaAPI.Models
{
    public enum StatusMatricula
    {
        Ativa = 1,
        Suspensa = 2,
        Cancelada = 3,
        Vencida = 4
    }

    public class Matricula
    {
        [Key]
        public int Id { get; set; }

        [Required(ErrorMessage = "Aluno é obrigatório")]
        public int AlunoId { get; set; }

        [Required(ErrorMessage = "Plano é obrigatório")]
        public int PlanoId { get; set; }

        [Required(ErrorMessage = "Professor responsável é obrigatório")]
        public int ProfessorId { get; set; }

        [Required(ErrorMessage = "Data de início é obrigatória")]
        public DateTime DataInicio { get; set; }

        public DateTime DataFim { get; set; }

        public StatusMatricula Status { get; set; } = StatusMatricula.Ativa;

        [MaxLength(500)]
        public string? Observacoes { get; set; }

        public DateTime CriadoEm { get; set; } = DateTime.UtcNow;


        [ForeignKey("AlunoId")]
        public Aluno? Aluno { get; set; }

        [ForeignKey("PlanoId")]
        public Plano? Plano { get; set; }

        [ForeignKey("ProfessorId")]
        public Professor? Professor { get; set; }
    }
}

File: projeto 2/academia/laravel-frontend/app/Models/Matricula.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Matricula extends Model
{
    use HasFactory;

    protected $fillable = [
        'aluno_id',
        'plano_id',
        'professor_id',
        'data_inicio',
        'data_fim',
        'status',
        'observacoes',
    ];

    protected $casts = [
        'data_inicio' => 'date',
        'data_fim'    => 'date',
    ];

    // Status possíveis
    const STATUS_ATIVA     = 'Ativa';
    const STATUS_SUSPENSA  = 'Suspensa';
    const STATUS_CANCELADA = 'Cancelada';
    const STATUS_VENCIDA   = 'Vencida';

    // Relacionamentos
    public function aluno()
    {
        return $this->belongsTo(Aluno::class);
    }

    public function plano()
    {
        return $this->belongsTo(Plano::class);
    }

    public function professor()
    {
        return $this->belongsTo(Professor::class);
    }

    // Scopes
    public function scopeAtivas($query)
    {
        return $query->where('status', self::STATUS_ATIVA);
    }

    public function scopeVencendo($query, $dias = 30)
    {
        return $query->where('status', self::STATUS_ATIVA)
                     ->where('data_fim', '<=', now()->addDays($dias));
    }
}
